src/Language/JavaScript/Parser/AST.hs

```diff
@@ -71,9 +73,10 @@ data JSStatement
     | JSSwitch !JSAnnot !JSAnnot !JSExpression !JSAnnot !JSAnnot ![JSSwitchParts] !JSAnnot !JSSemi -- ^switch,lb,expr,rb,caseblock,autosemi
     | JSThrow !JSAnnot !JSExpression !JSSemi -- ^throw val autosemi
     | JSTry !JSAnnot !JSBlock ![JSTryCatch] !JSTryFinally -- ^try,block,catches,finally
-    | JSVariable !JSAnnot !(JSCommaList JSExpression) !JSSemi -- ^var|const, decl, autosemi
+    | JSVariable !JSAnnot !(JSCommaList JSExpression) !JSSemi -- ^var, decl, autosemi
     | JSWhile !JSAnnot !JSAnnot !JSExpression !JSAnnot !JSStatement -- ^while,lb,expr,rb,stmt
     | JSWith !JSAnnot !JSAnnot !JSExpression !JSAnnot !JSStatement !JSSemi -- ^with,lb,expr,rb,stmt list
+    | JSExport !JSAnnot !JSExportBody !JSSemi -- ^export, body, autosemi
```

Comment:
Is introducing `JSExportBody` to support the following variants the right abstraction?

```
-- ExportDeclaration :                                                        See 15.2.3
-- [ ]    export * FromClause ;
-- [ ]    export ExportClause FromClause ;
-- [x]    export ExportClause ;
-- [x]    export VariableStatement
-- [ ]    export Declaration
-- [ ]    export default HoistableDeclaration[Default]
-- [ ]    export default ClassDeclaration[Default]
-- [ ]    export default [lookahead ∉ { function, class }] AssignmentExpression[In] ;
```

Q: What makes for the right AST abstraction?

Reply:
Not sure. You decide :)

Reply:
Decided to revisit this a bit and opened https://github.com/erikd/language-javascript/pull/80. Let me know what you think.